internal/batches/service/service.go

```diff
@@ -57,30 +59,41 @@ const sourcegraphVersionQuery = `query SourcegraphVersion {
 
 // getSourcegraphVersion queries the Sourcegraph GraphQL API to get the
 // current version of the Sourcegraph instance.
-func (svc *Service) getSourcegraphVersion(ctx context.Context) (string, error) {
+
+func (svc *Service) getSourcegraphVersionAndMaxChangesetsCount(ctx context.Context) (string, int, error) {
 	var result struct {
-		Site struct {
+		maxUnlicensedChangesets int
+		Site                    struct {
 			ProductVersion string
 		}
 	}
 
-	ok, err := svc.client.NewQuery(sourcegraphVersionQuery).Do(ctx, &result)
+	ok, err := svc.client.NewQuery(getInstanceInfo).Do(ctx, &result)
 	if err != nil || !ok {
-		return "", err
+		return "", 0, err
 	}
 
-	return result.Site.ProductVersion, err
+	//add to return
+	return result.Site.ProductVersion, result.maxUnlicensedChangesets, err
 }
 
-// DetermineFeatureFlags fetches the version of the configured Sourcegraph and
+// DetermineFeatureFlagsAndLicense fetches the version of the configured Sourcegraph and
 // returns the enabled features.
-func (svc *Service) DetermineFeatureFlags(ctx context.Context) (*batches.FeatureFlags, error) {
-	version, err := svc.getSourcegraphVersion(ctx)
+
+func (svc *Service) DetermineFeatureFlagsAndLicense(ctx context.Context) (*batches.LicenseRestrictions, *batches.FeatureFlags, error) {
```

Comment:
Nit: We should probably either call the function `DetermineLicenseAndFeatureFlags` or flip the order of the returned items: `(*batches.FeatureFlags, *batches.LicenseRestrictions, error)` so that they're self-consistent.